src/environmentd/src/http/sql.rs

```diff
@@ -194,62 +472,93 @@ async fn execute_request(
                 }
 
                 let stmt = stmts.pop().unwrap();
-                check_prohibited_stmts(&stmt)?;
+                check_prohibited_stmts(sender, &stmt)?;
 
                 stmt_groups.push(vec![(stmt, params)]);
             }
         }
     }
 
-    for stmt_group in stmt_groups {
+    'outer: for stmt_group in stmt_groups {
         let num_stmts = stmt_group.len();
         for (stmt, params) in stmt_group {
```

Comment:
I think extracting this entire block into some method like `execute_stmt_group` can help avoid the named `break`s.

Reply:
Done.